R/target_market_share.R

```diff
@@ -95,8 +93,23 @@ target_market_share <- function(data,
 
   check_input_for_crucial_columns(data, abcd, scenario)
 
+  abcd <- fill_and_warn_na(abcd, "production")
+  abcd <- dplyr::summarize(
+    abcd,
+    production = sum(.data[["production"]]),
+    .by = -"production"
+  )
+
   data <- aggregate_by_name_abcd(data)
 
+  if ("production" %in% colnames(scenario)) {
+    warn("The column `production` has been removed from the dataset `scenario`.
```

Comment:
how is this ever an issue?

Reply:
Not sure. I didn't add this check in this PR, i just refactored it from here: 
https://github.com/RMI-PACTA/r2dii.analysis/blob/170d4b0d60a3514607a962d32d2a848baaf298e9/R/target_market_share.R#L112-L118

It may be superfluous, but I guess when it was introduced there was some reason for it.

Reply:
Was introduced in this PR: https://github.com/RMI-PACTA/r2dii.analysis/pull/406

Reply:
Happy to explore if it's necessary, but I'd rather do so in a separate issue/ PR.